furms-web-ui/src/main/java/io/imunity/furms/ui/VaadinBroadcaster.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020 Bixbit s.c. All rights reserved.
+ * See LICENSE file for licensing information.
+ */
+
+package io.imunity.furms.ui;
+
+import com.vaadin.flow.shared.Registration;
+import io.imunity.furms.domain.communities.CommunityEvent;
+import io.imunity.furms.domain.projects.ProjectEvent;
+import io.imunity.furms.domain.sites.SiteEvent;
+import io.imunity.furms.domain.users.UserEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+@Component
+public class VaadinBroadcaster {
+	private final static Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+	private final List<Runnable> listeners = Collections.synchronizedList(new LinkedList<>());
+
+	public Registration register(Runnable listener) {
```

Comment:
We have one big problem here: there is no way to express interest in some specific events, and what's more listeners are not getting the event!

As soon as we will have another object listening, this completely won't work, right? And even now ViewPicker could be optimized, knowing what was changed. E.g. if project/community/site was removed and it is not in the picker currently, we can just skip the costly refreshing.